File: submission-code-server-database/src/main/java/fr/gouv/stopc/submission/code/server/database/entity/SubmissionCode.java
package fr.gouv.stopc.submission.code.server.database.entity;

import lombok.Data;

import javax.persistence.*;
import java.time.OffsetDateTime;


@Data
@Entity
@Table(name ="submission_code")
public class SubmissionCode {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    @Column(name = "lot")
    private long lot;

    @Column(name = "code", nullable = false, unique = true)
    private String code;

    @Column(name = "type_code", nullable = false)
    private String type;

    @Column(name = "date_end_validity")
    private OffsetDateTime dateEndValidity;

    @Column(name = "date_available")
    private OffsetDateTime dateAvailable;

    @Column(name = "date_use")
    private OffsetDateTime dateUse;

    @Column(name = "date_generation")
    private OffsetDateTime dateGeneration;

    @Column(name = "used")
    private Boolean used;

}

File: src/main/kotlin/fr/gouv/stopc/submissioncode/repository/model/SubmissionCode.kt
package fr.gouv.stopc.submissioncode.repository.model

import java.time.Instant
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "submission_code")
data class SubmissionCode(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    val id: Long = 0,

    @Column(name = "code")
    val code: String,

    @Column(name = "type_code")
    val type: String,

    @Column(name = "date_end_validity")
    val dateEndValidity: Instant,

    @Column(name = "date_available")
    val dateAvailable: Instant,

    @Column(name = "date_generation")
    val dateGeneration: Instant,

    @Column(name = "date_use")
    val dateUse: Instant? = null,

    @Column(name = "used")
    val used: Boolean
) {

    enum class Type(val dbValue: String) {
        LONG("1"),
        SHORT("2"),
        TEST("3")
    }
}
